samples/custom_dispatch/vulkan/shaders/example_udon.mlir

```diff
@@ -0,0 +1,75 @@
+// RUN:     --iree-preprocessing-transform-library=%p/udon_spec.mlir | \
+// RUN: iree-run-module \
+// RUN:     --device=vulkan \
+// RUN:     --module=- \
+// RUN:     --function=mixed_invocation \
+// RUN:     --input=8xf32=2 \
+// RUN:     --input=8xf32=4 | \
+// RUN: FileCheck %s
+
+// The configuration used for executable compilation.
+// This lets the compiler and runtime know the format and requirements of the
+// executable binaries produced and multiple variants with differing formats
+// and compilation options (architectures, etc) can be embedded for runtime
+// selection.
+// HACK: Currently this must match EXACTLY with the executable target for the
+// custom kernel. For things to be truly portable, we need to be able to compare
+// executable configurations.
+#spirv_target = #hal.executable.target<"vulkan", "vulkan-spirv-fb", {
+  spirv.target_env = #spirv.target_env<
+    #spirv.vce<v1.3, [Shader, GroupNonUniform, GroupNonUniformBallot], [SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers]>,
+    #spirv.resource_limits<max_compute_workgroup_size = [128, 128, 64], subgroup_size = 64>
+  >
+}>
+
+// The target devices that the program will run on. We can compile and run with
+// multiple targets, but this example is maintaining an implicit requirement
+// that the custom kernel being spliced in is supported by the target device,
+// hence we only support vulkan here. It is possible to hand author a custom
+// kernel that supports multiple targets by specifying an object per-target, but
+// that requires authoring the kernel for multiple targets.
+#vulkan_target = #hal.device.target<"vulkan", {
+  executable_targets = [#spirv_target],
+  // HACK: Vulkan target currently uses the legacy synchronous execution model.
+  legacy_sync
+}>
+
+#map = affine_map<(d0, d1) -> (d0, d1)>
+#map1 = affine_map<(d0, d1) -> (d0)>
+module @example attributes {hal.device.targets = [#vulkan_target]} {
+
+  // CHECK-LABEL: EXEC @mixed_invocation
+  func.func @mixed_invocation(%arg0: tensor<1x1024xf32>, %arg1: tensor<1x1024xf32>) -> tensor<1xi64> {
+    // Code gen some other ops - these will interleave with the matched and
+    // replaced ones but naturally won't be able to fuse with them.
+    %add = arith.addf %arg0, %arg1 : tensor<1x1024xf32>
+
+    %c0_i64 = arith.constant 0 : i64
+    %cst = arith.constant 0xFF800000 : f32
+    %1 = tensor.empty() : tensor<1xi64>
+    %2 = linalg.fill ins(%c0_i64 : i64) outs(%1 : tensor<1xi64>) -> tensor<1xi64>
+    %3 = tensor.empty() : tensor<1xf32>
+    %4 = linalg.fill ins(%cst : f32) outs(%3 : tensor<1xf32>) -> tensor<1xf32>
+    // Argmax that is the target for the custom kernel. Note that this operation
+    // only has uses for a single result and takes a single input. More complex
+    // subgraph matching is possible but as of writing this comment, the
+    // transform dialect has a number of footguns related to how matching works.
```

Comment:
We need to nail this down a bit. We will need more complex matching.

Reply:
I'll add an example with a more complex matching separately, but I'm probably projecting my past experiences a bit here...